### DISCUSSION TITLE
Async example does not mention x-data

### DISCUSSION THREAD

**Discussion by Youdaman** [alpinejs/alpine#4399] [7. Bugs (archived)]
In the docs for Async [here](https://alpinejs.dev/advanced/async) it could/should mention `x-data`, for example:
```html
<span x-data x-text="getLabel()"></span>
```

I see that the [page about `x-data`](https://alpinejs.dev/directives/data) says:

> Everything in Alpine starts with the x-data directive.

However as a newcomer to the framework I didn't read that page prior to reading the docs for Async and so could not get the example there working until I saw mention `x-data` in an issue where someone was having the same problem.

**Comment by SimoTod** [comment]
Most of the documentation is in the repo. The page you mentioned is available at https://github.com/alpinejs/alpine/blob/main/packages/docs/src/en/advanced/async.md

For this page, given it's supposed to show a fully working example, it probably makes enough sense to add it.

**Comment by ekwoka** [comment]
Every directive page was recently updated to include these notes.

Like here: https://alpinejs.dev/directives/for#iterating-over-a-range
